samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -141,6 +143,9 @@ object SamzaContainer extends Logging {
     val systemConfig = new SystemConfig(config)
     val containerModel = jobModel.getContainers.get(containerId)
     val containerName = "samza-container-%s" format containerId
+    val jobName = config.getName.getOrElse(throw new ConfigException("Missing required config: job.name"))
+    val jobId = config.getJobId
+    val execEnvContainerId = System.getenv(ShellCommandConfig.ENV_EXECUTION_ENV_CONTAINER_ID)
```

Comment:
Maybe pass as a constructor param? Would be good to remove system properties dependency in SamzaContainer. E.g., What happens in standalone?